web/packages/teleterm/src/services/tshd/createClient.ts

```diff
@@ -815,7 +815,32 @@ export default function createClient(
               if (err) {
                 reject(err);
               } else {
-                resolve(res.toObject());
+                resolve({
+                  nextKey: res.getNextKey(),
+                  resources: res.getResourcesList().map(p => {
+                    switch (p.getResourceCase()) {
+                      case api.PaginatedResource.ResourceCase.SERVER:
+                        return {
+                          kind: 'server',
+                          resource: p.getServer().toObject() as types.Server,
+                        };
+                      case api.PaginatedResource.ResourceCase.DATABASE:
+                        return {
+                          kind: 'database',
+                          resource: p
+                            .getDatabase()
+                            .toObject() as types.Database,
+                        };
+                      case api.PaginatedResource.ResourceCase.KUBE:
+                        return {
+                          kind: 'kube',
+                          resource: p.getKube().toObject() as types.Kube,
+                        };
+                      default:
+                        throw new Error('resource not set');
```

Comment:
This will blow up if, say, v15 Connect connects to v16 cluster (a version combo supported by us) and the cluster returns some new kind of resource that v15 Connect doesn't know about. The client code seems to have the same problem, I'll ask Michael about this.

In protobuf enums, typically the unspecified value of the enum is used for that, see [my question from a year ago](https://gravitational.slack.com/archives/C08HF8E9F/p1659351417856139). However, this would force us to support some kind of an extra kind of resource in the UI. Alternatively, I suppose we could just filter out those resources and log the fact that we ignored an unsupported resource?

Reply:
This would indeed blow up if the server returns a type that the client isn't ready to handle. We decided to send a set of default kinds for now (the server will accept any request with no kinds filter). I've updated the [web UI to send a set of default kinds](https://github.com/gravitational/teleport/pull/33314) now. 

In the future, we will make a set of kinds an explicit requirement in requests.

Reply:
Thanks for catching this, I added filtering and logging.